File: Constructive algorithms/codeforces_1930B.py

# Problem link ===>> https://codeforces.com/problemset/problem/####/A


def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        st = set(range(1, n + 1))

    while st:
        print(min(st), end=" ")
        st.remove(min(st))

        if st:
            print(max(st), end=" ")
            st.remove(max(st))

        print()
        


if __name__ == "__main__":
    main()

File: Constructive algorithms/codeforces_1930B.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/1930/B
// submission link ===>> https://codeforces.com/contest/1930/submission/261343765

#include <bits/stdc++.h>
#define ll long long
using namespace std;

int main()
{
    ll t;
    cin >> t;
    while (t--)
    {
        ll n;
        cin >> n;

        set<ll> st;
        for (ll i = 1; i <= n; i++)
            st.insert(i);

        for (ll i = 1; i <= n; i++)
        {
            if (i % 2)
            {
                cout << *st.begin() << " ";
                st.erase(st.begin());
            }
            else
            {
                cout << *st.rbegin() << " ";
                st.erase(--st.end());
            }
        }

        cout << endl;
    }
    return 0;
}